This patch removes the unneeded gpu_clk driver as it fails
to initialize anyway. It looks like the functionality of this
driver is built into the mali driver itself.

diff -ur original/arch/arm64/boot/dts/amlogic/mesongxm-gpu-t82x.dtsi modified/arch/arm64/boot/dts/amlogic/mesongxm-gpu-t82x.dtsi
--- original/arch/arm64/boot/dts/amlogic/mesongxm-gpu-t82x.dtsi	2017-05-15 16:45:50.000000000 +0300
+++ modified/arch/arm64/boot/dts/amlogic/mesongxm-gpu-t82x.dtsi	2017-05-27 00:51:08.239410884 +0300
@@ -118,91 +118,4 @@
 		  };
 	};
 
-	gpu_clk:gpu_clk@c883c00 {
-	    compatible = "meson, gpu-clkgen-1.00.a";
-		#clock-cells = <0>;
-        reg = <0 0xc883c000 0 0x001000>;
-		clocks = <&clock CLK_FPLL_DIV3>,
-			<&clock CLK_FPLL_DIV4>,
-			<&clock CLK_FPLL_DIV5>,
-			<&clock CLK_FPLL_DIV7>,
-			<&clock GP0_PLL>,
-			<&clock CLK_GPU>,
-			<&clock CLK_GPU_0>,
-			<&clock CLK_GPU_1>;
-		clock-names =
-			"fclk_div3",
-			"fclk_div4",
-			"fclk_div5",
-			"fclk_div7",
-			"gp0_pll",
-			"clk_gpu",
-			"clk_gpu_0",
-			"clk_gpu_1";
-        clock-output-names = "clk_mali";
-		tbl = <&t82x_clk125_cfg  &t82x_clk250_cfg &t82x_clk285_cfg &t82x_clk400_cfg &t82x_clk500_cfg &t82x_clk666_cfg>;
-		  t82x_clk125_cfg:clk125_cfg {
-			clk_freq = <125000000>;
-			clk_parent = "fclk_div4";
-			clkp_freq = <500000000>;
-			voltage = <1150>;
-			keep_count = <5>;
-			threshold = <30 120>;
-		  };
-
-		  t82x_clk250_cfg:clk250_cfg {
-			clk_freq = <250000000>;
-			clk_parent = "fclk_div4";
-			clkp_freq = <500000000>;
-			voltage = <1150>;
-			keep_count = <5>;
-			threshold = <80 170>;
-		  };
-
-		  t82x_clk285_cfg:clk285_cfg {
-			clk_freq = <285714000>;
-			clk_parent = "fclk_div7";
-			clkp_freq = <285714000>;
-			voltage = <1150>;
-			keep_count = <5>;
-			threshold = <100 190>;
-		  };
-
-		  t82x_clk400_cfg:clk400_cfg {
-			clk_freq = <400000000>;
-			clk_parent = "fclk_div5";
-			clkp_freq = <400000000>;
-			voltage = <1150>;
-			keep_count = <5>;
-			threshold = <152 207>;
-		  };
-
-		  t82x_clk500_cfg:clk500_cfg {
-			clk_freq = <500000000>;
-			clk_parent = "fclk_div4";
-			clkp_freq = <500000000>;
-			voltage = <1150>;
-			keep_count = <5>;
-			threshold = <180 220>;
-		  };
-
-		  t82x_clk666_cfg:clk666_cfg {
-			clk_freq = <666666000>;
-			clk_parent = "fclk_div3";
-			clkp_freq = <666666000>;
-			voltage = <1150>;
-			keep_count = <5>;
-			threshold = <210 236>;
-		  };
-
-		  t82x_clk720_cfg:clk720_cfg {
-			clk_freq = <720000000>;
-			clk_parent = "gp0_pll";
-			clkp_freq = <720000000>;
-			voltage = <1150>;
-			keep_count = <5>;
-			threshold = <230 255>;
-		  };
-	};
-
 };/* end of / */
